pkg/pool-weighted/contracts/managed/ManagedPool.sol

```diff
@@ -225,29 +263,35 @@ contract ManagedPool is ManagedPoolSettings {
         uint256[] memory balances,
         bytes memory userData
     ) internal virtual override returns (uint256 bptAmountOut, uint256[] memory amountsIn) {
-        (IERC20[] memory tokens, ) = _getPoolTokens();
+        // The Vault passes an array of balances which includes the pool's BPT (This always sits in the first position).
+        // We want to separate this from the other balances before continuing with the join.
+        uint256 virtualSupply;
+        (virtualSupply, balances) = ComposablePoolLib.dropBptFromBalances(totalSupply(), balances);
```

Comment:
I've always hated this quirk about solidity

Reply:
Yeah, I hate we can't mix assignments and definitions like this.